pages/[id]/index.js

```diff
@@ -1,16 +1,64 @@
 import initialPictures from "@/store";
 import PostingDetails from "@/components/PostingDetails";
 import { useRouter } from "next/router";
+import CommentForm from "@/components/CommentForm";
+import { useState } from "react";
+import { uid } from "uid";
+import CommentList from "@/components/CommentList";
 
 export default function PostDetailsPage() {
+  const [pictures, setPictures] = useState(initialPictures);
   const router = useRouter();
   const { id } = router.query;
 
-  const picture = initialPictures.find((picture) => {
+  // getting the requested picture
+  const selectedPicture = pictures.find((picture) => {
     return picture.id === id;
   });
 
-  if (picture) {
-    return <PostingDetails picture={picture} />;
+  // adding comment
+  function handleAddComment(comment) {
+    setPictures(
+      pictures.map((picture) => {
+        if (selectedPicture.id === picture.id) {
+          return {
+            ...picture,
+            comments: [
+              ...picture.comments,
+              { commentId: uid(), commentContent: comment },
+            ],
+          };
+        }
+        return picture;
+      })
+    );
+  }
+
+  // delete comment
+  function handleDeleteComment(deleteComment) {
+    setPictures(
+      pictures.map((picture) => {
+        if (selectedPicture.id === picture.id) {
+          const filteredComments = picture.comments.filter((comment) => {
+            return comment.commentId !== deleteComment.commentId;
+          });
+          picture.comments = filteredComments;
+        }
+        return picture;
+      })
+    );
+  }
```

Comment:
That doesn't look ideal to me, let's talk about that for a minute, I think that's easier in this case.

I'll get to you in a minute.

Reply:
Alright. Let me know